### PR TITLE
 Add netifd script to control SSTP connections.

### PR DESCRIPTION
**Pull request by rkkoszewski**
Description: Add netifd script to control SSTP connections.

Script was made by harveyhu2012 and slightly modified by me (rkkoszewski)

It is required for luci-proto-sstp (https://github.com/openwrt/luci/pull/2768)

### CODE REVIEW

File: net/sstp-client/files/lib/netifd/proto/sstp.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		ip-up-script /lib/netifd/ppp-up \
+		ipv6-up-script /lib/netifd/ppp-up \
+		ip-down-script /lib/netifd/ppp-down \
+		ipv6-down-script /lib/netifd/ppp-down \
+		${mtu:+mtu $mtu mru $mtu} \
+		$pppd_options
+		
+	# WORKAROUND: Workaround to properly register the sstp interface (As seeen in: https://forum.archive.openwrt.org/viewtopic.php?id=58007)
+	# WORKAROUND: Start
+	sleep 10
```
Comment (rkkoszewski) [inline net/sstp-client/files/lib/netifd/proto/sstp.sh]: Anybody has a better idea as an approach for this? Setting a fixed timeout seems hacky at best. From what I understand, the reason this fails without the timeout is that the sstpc starts a pppd subprocess asyncronously and if the next commands get invoked before it started, they fail to register the new device. 

I don't know if it's somehow possible to get the PID of the currently running proto_run_command, if yes, then pgrep could be used to check for the subprocess every second or so, or if the proto_run_command process dies then just abort. But so far I was unable to find the PID of the proto_run_command programatically. Anybody has an idea?

**Comment by rkkoszewski** [comment]
Anybody can suggest what to do regarding the missing signoff? The original author @harveyhu2012 doesn't seem to be active on GitHub since 2014.

**Comment by neheb** [comment]
Needs a PKG_RELEASE bump. Also, please rebase against master as there's a random merge commit in there.

**Comment by rkkoszewski** [comment]
> Needs a PKG_RELEASE bump. Also, please rebase against master as there's a random merge commit in there.

Rebased against master and bumped the package version. Haven't yet worked with the rebase command and I added the workaround from my commit to the first one by mistake, therefore my workaround commit appears as just a space (But it's signed off by me). Squashing should fix that issue though.

**Comment by neheb** [comment]
Maintainer is @fededim

File: net/sstp-client/Makefile [hunk 2]
```diff
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=sstp-client
-PKG_VERSION:=1.0.12
+PKG_VERSION:=1.1.0
```
[the remark this answers is not in the document]
Comment (rkkoszewski) [reply net/sstp-client/Makefile]: Hi @fededim, sorry but I mistook PKG_RELEASE with PKG_VERSION. I will correct it in the next push.

[the remark this answers is not in the document]
Comment (rkkoszewski) [reply net/sstp-client/Makefile]: Ok, so this got fixed in this commit now: https://github.com/openwrt/packages/pull/9221/commits/045149cb217452d04c13bf6d1afdfcc388fc52b6

[the remark this answers is not in the document]
Comment (fededim) [reply net/sstp-client/Makefile]: Ok changes are fine for me

[the remark this answers is not in the document]
Comment (fededim) [reply net/sstp-client/Makefile]: Remember to squash and amend your commits, otherwise you can't merge

**Comment by rkkoszewski** [comment]
Is there still something missing in this PR preventing it from being merged? @neheb @fededim

**Comment by neheb** [comment]
Maintainer approved. DCO is failing because of mismatching author name. Will merge anyway.

**Comment by BKPepe** [comment]
@neheb I think we have those checks for a reason. The DCO can be skipped, so CircleCI can try to build it. Never merge anything when there are 0 of 2 checks passed. 

Also, commits should be squashed as maintainer requested.

**Comment by rkkoszewski** [comment]
Hi @BKPepe, sorry I thought that GitHub does the squashing automatically (Was my first contribution here, although it seems like it did squash it from what I can see in the commits). Is there any advantage of performing it manually?

**Pull request merged**